### ISSUE TITLE
[$250][HOLDnecolas/react-native-web#2709] Tab - When switching tabs, the Custom Profile Avatar shows delayed

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#45853]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.10-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/45663
## Action Performed:
PreCondition: User has a custom avatar added to their profile.
1. Access staging.new.expensify.com
2. Sign into a valid account (Beta enabled)
3. Go to inbox tab
4. Go to Search tab
5. Go to Profile tab

## Expected Result:
User expects the Avatar image to load instantly upon changing tabs

## Actual Result:
The avatar image blinks a takes a couple of seconds to load

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/a44efc0b-4e34-4df1-88bf-d1ac53d9aa60

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ec9d563c4ff0e779</li>
        <li>Upwork Job ID: 1816230532262348738</li>
        <li>Last Price Increase: 2024-08-07</li>
<li>Automatic offers: </li>
<ul>
<li>dominictb | Contributor | 103496599</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @s77rt</details>

**Comment by lanitochka17** [comment]
@strepanier03  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by strepanier03** [comment]
Working on this now.

**Comment by strepanier03** [comment]
Repro'd all good here.

**Comment by s77rt** [comment]
@dominictb 

> in order to land in that state when the component mount, the image should be in the cache

Isn't the image being cached?

**Comment by s77rt** [comment]
@dominictb After an image is loaded the first time it should be cached. Next time you render the same image it should load from the cache. Why this is not the case here?

**Comment by s77rt** [comment]
@dominictb Sounds to me like a bug in RNW. If we overwrite the [initial state](https://github.com/necolas/react-native-web/blob/54c14d64dabd175e8055e1dc92e9196c821f9b7d/packages/react-native-web/src/exports/Image/index.js#L203-L212) to be `LOADED`. Will the image load correctly on tab switch?

**Comment by s77rt** [comment]
@dominictb Will the image load correctly (from cache) in that case? (regardless of the placeholder)

**Comment by s77rt** [comment]
@dominictb In that case the bug should be fixed at RNW level

**Comment by s77rt** [comment]
@dominictb Adding the image uri to the cache entries makes sense to me but a better solution is to make use of the browser cache, that is to modify the [`ImageUriCache.has`](https://github.com/necolas/react-native-web/blob/54c14d64dabd175e8055e1dc92e9196c821f9b7d/packages/react-native-web/src/modules/ImageLoader/index.js#L16-L20) method to check if a uri is already cached. Can you look into that?

**Comment by s77rt** [comment]
@dominictb Checking the existing cache is more reliable than maintaining a second one. I found this solution https://stackoverflow.com/a/50111407 that may work. Can you check it out?

**Comment by s77rt** [comment]
@dominictb Why we have a globally defined Image? Also please make sure if an image does not exist in cache we don't fetch it.

> instead of check using `ImageLoader.has`, which under the hood use `ImageUriCache` to check, we can use `ImageLoader.isCached`

We don't want to replace `ImageLoader.has`. Instead that method should make use of new `isCached` mechanism

**Comment by s77rt** [comment]
@dominictb `ImageUriCache.has` is what we should update. 

> Also please make sure if an image does not exist in cache we don't fetch it.

Can you look at that as well?

**Comment by s77rt** [comment]
@dominictb 

> we can add an early return here if `state` is `LOADED`

This is unrelated to the new cache function. Notice that after creating a `Image` and setting the `src`, the browser fill fetch that image which is an unwanted behaviour

**Comment by s77rt** [comment]
Still looking for proposals

**Comment by s77rt** [comment]
@dominictb I don't see how your are proposing to fix the unnecessary fetch when checking for cache. Can you please restate your solution in plain English? - This should be related only to `ImageLoader.has` and not to the logic that loads the image, i.e. calling `ImageLoader.has` should not make a fetch request.

**Comment by s77rt** [comment]
@dominictb 

> document.appendChild(image) // if the image is in the cache -> display the image using DOM manipulation

Why are you trying to display the image? We just want to check if the image is in cache or not. Also isn't unsetting the src helpful?

**Comment by s77rt** [comment]
@dominictb The solution looks good to me. Can you please raise a PR in RNW. Please provide a reproducible example and follow the repo's contribution guide.

**Comment by s77rt** [comment]
@dominictb At this point I don't think the order matters much.

🎀 👀 🎀 C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/45853#issuecomment-2245819918)

**Comment by s77rt** [comment]
Not overdue. @dominictb is going to raise a PR upstream (RNW)

**Comment by s77rt** [comment]
Not overdue. Waiting on RNW PR feedback

**Comment by s77rt** [comment]
Same ^

**Comment by s77rt** [comment]
@stitesExpensify Let's make this `Weekly` and hold on https://github.com/necolas/react-native-web/pull/2709

**Comment by s77rt** [comment]
Still on hold

**Comment by strepanier03** [comment]
Put this in #quality.

**Comment by s77rt** [comment]
@dominictb Seems like the upstream PR needs your attention. Please provide a reproducer for the repo maintainers to test

**Comment by s77rt** [comment]
@dominictb Is the bug no longer reproducible? Did you try with a custom avatar?

**Comment by s77rt** [comment]
@dominictb If we have nothing to fix then let's do nothing
